### PR TITLE
Test PR to make sure that Travis runs properly

### PR DESCRIPTION
**Pull request by prestonlimlianjie**
Terraform x Travis deployment pipeline

1. Automate travis deployment of terraform
- Set up the Travis pipeline 
- Encrypted GCP service account credentials for *isomer-travis* in the repo
- Changed preston-terraform and pallani-terraform service account credentials to Viewer
- terraform plan runs when a PR is made to *master*
- terraform apply -auto-approve runs when the PR is merged into *master*

2. Added more Terraformed components:
- Added AWS CDN
- Added GCP IAM
- Added GCP Cloud Bucket

### CODE REVIEW

File: run-terraform.sh [hunk 1]
```diff
@@ -1,18 +1,21 @@
 #!/bin/bash
 
-cd credentials
-. ./credentials.sh
-cd ..
-
-cd run-dir
-rm *.tf
-cd ..
+if [ ! -d "run-dir" ]; then
```
Comment (pallani) [inline run-terraform.sh]: Add a brief comment on what this block is doing.

File: run-terraform.sh [hunk 2]
```diff
@@ -1,18 +1,21 @@
...
+  	mkdir run-dir
+else
+  	cd run-dir
+  	rm *.tf
+  	cd ..
+fi
 
 for DIR in terraform/*; do
```
Comment (pallani) [inline run-terraform.sh]: Same as above. Add a brief comment on what this block is doing.

File: script/deploy [hunk 3]
```diff
@@ -0,0 +1,3 @@
+#!/bin/bash
+
+./terraform apply -auto-approve
```
Comment (pallani) [inline script/deploy]: Add comment on what auto-approve does.

**Comment by pallani** [comment]
Do we need to include the terraform binary? Should it be installed in travis?

**Comment by pallani** [comment]
@prestonlimlianjie Think of the comments as helping someone who comes to look at our code to figure how we are using terraform.

File: .travis.yml [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+  - name: test
+    if: branch = master AND \
+        type = pull_request
+  - name: deploy
+    if: branch = master AND \
+        type = push
+jobs:
+  include:
+    - stage: test
+      script: ./script/test
```
[the remark this answers is not in the document]
Comment (prestonlimlianjie) [reply .travis.yml]: Good call - made the requested changes

File: run-terraform.sh (hunk 1 above)
Comment (prestonlimlianjie) [reply run-terraform.sh]: Done

File: run-terraform.sh (hunk 2 above)
Comment (prestonlimlianjie) [reply run-terraform.sh]: Done

File: script/deploy (hunk 3 above)
Comment (prestonlimlianjie) [reply script/deploy]: Done

**Comment by prestonlimlianjie** [comment]
> Do we need to include the terraform binary? Should it be installed in travis?

It's currently stored in the GitHub repo so we don't have to reinstall it each time Travis runs

**Pull request merged**